packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart

```diff
@@ -203,6 +189,11 @@ class FirebaseAuthenticationService {
     } on FirebaseAuthException catch (e) {
       log?.e(e);
       return await _handleAccountExists(e);
+    } on FacebookAuthException catch (e) {
+      log?.e(e);
+      if(e.errorCode == FacebookAuthErrorCode.CANCELLED) return null;
```

Comment:
This library will always return a `FirebaseAuthenticationResult`. Please return `FirebaseAuthenticationResult.error(errorMessage: 'Facebook login has been cancelled by the user');` instead of null.

Reply:
If we don't want to return null here I can just remove `if(e.errorCode == FacebookAuthErrorCode.CANCELLED) return null;` and `return FirebaseAuthenticationResult.error(errorMessage: e.toString());` will handle it.

A question about Google and Apple sign in, they both have return null on cancel:

Google:
```
      if (googleSignInAccount == null) {
        log?.i('Process is canceled by the user');
        return null;
      }
```
Apple:
```
    on SignInWithAppleAuthorizationException catch (e) {
      if (e.code == AuthorizationErrorCode.canceled) return null;

      return FirebaseAuthenticationResult.error(errorMessage: e.toString());
    } 
```

Maybe this should be changed in another PR?

Reply:
Good catch. We can probably change that in this PR as well if you don't mind.